tests/xharness/Jenkins/TestSelector.cs

```diff
@@ -150,14 +162,9 @@ void SelectTestsByModifiedFiles (int pullRequest)
 			foreach (var f in files)
 				MainLog.WriteLine ("    {0}", f);
 			
-			SetEnabled (files, mtouchPrefixes, "mtouch", ref jenkins.IncludeMtouch);
-			SetEnabled (files, mmpPrefixes, "mmp", ref jenkins.IncludeMmpTest);
-			SetEnabled (files, bclPrefixes, "bcl", ref jenkins.IncludeBcl);
-			SetEnabled (files, btouchPrefixes, "btouch", ref jenkins.IncludeBtouch);
-			SetEnabled (files, macBindingProject, "mac-binding-project", ref jenkins.IncludeMacBindingProject);
-			SetEnabled (files, xtroPrefixes, "xtro", ref jenkins.IncludeXtro);
-			SetEnabled (files, cecilPrefixes, "cecil", ref jenkins.IncludeCecil);
-			SetEnabled (files, dotnetPrefixes, "dotnet", ref jenkins.IncludeDotNet);
+			foreach (var selection in selectionPrefixesMap.Keys) {
```

Comment:
`all` should be processed first, so that doing `skip-all-tests,run-mtouch-tests` runs the mtouch tests (and nothing else). Just iterating over the keys do not guarantee any order, since this is a dictionary.